fire/SFMainMod.F90

```diff
@@ -540,15 +725,29 @@ subroutine rate_of_spread ( currentSite )
 
        ! write(fates_log(),*) 'ir',gamma_aptr,moist_damp,SF_val_fuel_energy,SF_val_miner_damp
 
+     
        if (((currentPatch%fuel_bulkd) <= 0.0_r8).or.(eps <= 0.0_r8).or.(q_ig <= 0.0_r8)) then
           currentPatch%ROS_front = 0.0_r8
-       else ! Equation 9. Thonicke et al. 2010. 
+          ROS_torch = 0.0_r8   
+       else ! Eq 9. Thonicke et al. 2010. 
             ! forward ROS in m/min
           currentPatch%ROS_front = (ir*xi*(1.0_r8+phi_wind)) / (currentPatch%fuel_bulkd*eps*q_ig)
           ! write(fates_log(),*) 'ROS',currentPatch%ROS_front,phi_wind,currentPatch%effect_wspeed
           ! write(fates_log(),*) 'ros calcs',currentPatch%fuel_bulkd,ir,xi,eps,q_ig
+
+          ! calculate heat release per unit area (HPA)(kJ/m2), Eq 2 Scott & Reinhardt 2001
+          ! and residence time (min), Eq 3 Scott & Reinhardt 2001
+          time_r = 12.595 / currentPatch%fuel_sav 
+          heat_per_area = ir * time_r          
+
+          ! calculate torching index based on wind speed and crown fuels 
+          ! ROS for crown torch initation (m/min), Eq 18 Scott & Reinhardt 2001
+          ! TODO slevis: Are we missing a -phi_s after the -1?
+          ROS_torch = 1._r8 / (54.683_r8 * wind_reduce) * &
+    ((60._r8 * passive_crown_FI * currentPatch%fuel_bulkd * eps * q_ig / &
+      (heat_per_area * ir * xi) - 1._r8) / (c * beta_ratio**-e))**(1._r8 / b)
```

Comment:
Moving post from https://github.com/slevisconsulting/slevis_fates_work/pull/1

@slevisconsulting wrote
`are we missing a -phi_s term after the -1`
@jkshuman replied
`that is for the slope. I meant to add it in as a placeholder with a value of zero as we do not have slope currently. So you can add a parameter "slope_factor" and set it to zero.`

Reply:
Thanks, @jkshuman